modules/navier_stokes/src/bcs/INSADSurfaceTensionBC.C

```diff
@@ -0,0 +1,42 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSADSurfaceTensionBC.h"
+
+registerMooseObject("NavierStokesApp", INSADSurfaceTensionBC);
+
+InputParameters
+INSADSurfaceTensionBC::validParams()
+{
+  InputParameters params = ADVectorIntegratedBC::validParams();
+  params.addClassDescription("Surface tension stresses.");
+  params.addParam<bool>("include_gradient_terms",
+                        false,
+                        "If the surface tension should include the gradient terms (increases "
+                        "fidelity, decreases stability)");
+  return params;
+}
+
+INSADSurfaceTensionBC::INSADSurfaceTensionBC(const InputParameters & parameters)
+  : ADVectorIntegratedBC(parameters),
+    _surface_term_curvature(getADMaterialProperty<RealVectorValue>("surface_term_curvature")),
+    _surface_term_gradient1(getADMaterialProperty<RealVectorValue>("surface_term_gradient1")),
+    _surface_term_gradient2(getADMaterialProperty<RealVectorValue>("surface_term_gradient2")),
+    _include_gradient_terms(getParam<bool>("include_gradient_terms"))
+{
+}
+
+ADReal
+INSADSurfaceTensionBC::computeQpResidual()
+{
+  auto force = _surface_term_curvature[_qp];
+  if (_include_gradient_terms)
+    force += _surface_term_gradient1[_qp] + _surface_term_gradient2[_qp];
+  return _test[_i][_qp] * force;
```

Comment:
I feel like this looks a lot like the kernel I had a long time ago, but my memory could be serving me wrong. Did you dig this up from git history or is this all from your own brain?

Reply:
No, we talked about the gradient terms and the Marangoni effect. You gave me a commit from a long time ago (around 2019 I think). I checked the Aria laser melt pool paper and they have these terms (curvature term and the gradient terms). The Aria material already had the gradient terms but not the curvature term. So I added that. I can check your commit from back then and if it is exactly this I can just pull that object back instead of creating a new node in git.

Reply:
Oh no don't worry about that. I was only curious